src/TLuaInterpreterMapper.cpp

```diff
@@ -3479,12 +3484,32 @@ int TLuaInterpreter::setRoomArea(lua_State* L)
         return warnArgumentValue(L, __func__, "no map present or loaded");
     }
 
-    const int id = getVerifiedInt(L, __func__, 1, "roomID");
-    if (!host.mpMap->mpRoomDB->getRoomIDList().contains(id)) {
-        return warnArgumentValue(L, __func__, csmInvalidRoomID.arg(id));
+    QVector<int> roomIds;
+    if (lua_isnumber(L, 1)) {
+        const int id = getVerifiedInt(L, __func__, 1, "roomID");
+        if (!host.mpMap->mpRoomDB->getRoomIDList().contains(id)) {
+            return warnArgumentValue(L, __func__, csmInvalidRoomID.arg(id));
+        }
+        roomIds.append(id);
```

Comment:
⚠️ Beware - there is nothing to stop the user providing the same room Id more than once - and if they do - and one of them is the *last* one provided - then the check later on to only do the area extremes recalculation on the last room will be triggered multiple times. OTOH **this** should not be harmful but could be avoided with a `std::set` rather than a `std::vector` which prohibits duplicates (and is sorted as a side effect)...

Reply:
I'll check this

Reply:
Updated.